clar2wasm/src/standard/standard.wat

```diff
@@ -1409,29 +1409,81 @@
     )
 
     (func $stdlib.sha256-buf (param $offset i32) (param $length i32) (param $offset-result i32) (result i32 i32)
+        ;; For binary representation, you can take a look at https://sha256algorithm.com/
+        ;; For initial values, constants and high level understanding of SHA-512, please take a look at this Rust implementation
+        ;; https://github.com/dandyvica/sha/blob/master/src/sha256.rs
+        ;; If you're interested in the paper, please take a look at https://eprint.iacr.org/2010/548.pdf
+
+        ;; High Level Overview
+        ;; 1. Params
+        ;;      a. We need $offset where we stored/wrote the data to be hashed.  
+        ;;      b. We need $length of the data to be hashed.  
+        ;;      c. We need $offset-result where we're going to store/write the final hash. 
+        ;;         Typically greater than $stack-pointer + total number of bytes used in the whole process
```

Comment:
Not true. Since this is a fixed amount of byte, this is computed during the wasm generation.

Reply:
That might not be true, but I'm a bit confused here. Are those fixed amount of bytes are `END_OF_STANDARD_DATA`? If so then `$stack-pointer` must be greater than that. And bytes used in the process are (initial values + bytes used to calculate words + length of the data to be hashed). Am I missing something?

Reply:
Oof, yes you are confused.
This is the offset at which you will find the hash, which should also be the result of that function call. It is `offset_res` in this function: https://github.com/stacks-network/clarity-wasm/blob/e2afe14d2aa199c042704b2569fff34fd36b6f03/clar2wasm/src/words/hashing.rs#L7
At `$stack-pointers`, we have free space that we can use as temporary for the computations.

Reply:
Comment Removed